src/node_buffer.cc

```diff
@@ -636,6 +633,11 @@ void StringWrite(const FunctionCallbackInfo<Value>& args) {
   if (offset >= ts_obj_length)
     return env->ThrowRangeError("Offset is out of bounds");
 
+  if (StringBytes::IsValidString(env->isolate(), str, encoding) == false) {
+    if (encoding == HEX)
+      return env->ThrowTypeError("Invalid hex string");
+  }
```

Comment:
Seems rather wasteful to call IsValidString() and only act on the result when encoding == HEX.